evm/src/generation/mod.rs

```diff
@@ -0,0 +1,57 @@
+use plonky2::field::extension::Extendable;
+use plonky2::field::polynomial::PolynomialValues;
+use plonky2::field::types::Field;
+use plonky2::hash::hash_types::RichField;
+
+use crate::all_stark::AllStark;
+use crate::cpu::bootstrap_kernel::generate_bootstrap_kernel;
+use crate::cpu::columns::NUM_CPU_COLUMNS;
+use crate::generation::state::GenerationState;
+use crate::util::trace_rows_to_poly_values;
+
+mod memory;
+pub(crate) mod state;
+
+pub type RlpMerkleProof = Vec<Vec<u8>>;
```

Comment:
nit: Maybe add a comment explaining how this is structured. Intuitively I would have thought the whole Merkle proof would be RLP encoded.

Reply:
Good point, added a comment. Packing them in a single RLP blob could work also, but one RLP blob per node seems to be conventional. I think the convention comes from EIP-1186 (it wasn't formally accepted, but some clients seem to support `eth_getProof` based on it anyway).